sdk/storage/azure-storage-blob/tests/test_blob_access_conditions.py

```diff
@@ -285,6 +285,24 @@ def test_delete_container_with_if_unmodified_fail(self, resource_group, location
 
         self.assertEqual(StorageErrorCode.condition_not_met, e.exception.error_code)
 
+    @GlobalStorageAccountPreparer()
+    def test_multi_put_blob_contains_headers(self, resource_group, location, storage_account, storage_account_key):
+        def _validate_headers(request):
+            header = request.http_request.headers.get('x-ms-meta-customheader')
+            self.assertEqual(header, 'test_value')
+
+        bsc = BlobServiceClient(
+            self.account_url(storage_account, "blob"), storage_account_key, max_single_put_size=100)
+        self._setup()
+        data = self.get_random_bytes(2 * 100)
+        self._create_container(self.container_name, bsc)
+        blob = bsc.get_blob_client(self.container_name, "blob1")
+        blob.upload_blob(
+            data,
+            metadata={'customheader': 'test_value'},
```

Comment:
what this pr is trying to solve is add customized header to each stage_block request:
```
blob.upload_blob(data, headers={'x-custom-header':'value'})
```
So you want to test if this customized header is there.
metadata isn't supposed to be set for stage_block operation, https://docs.microsoft.com/en-us/rest/api/storageservices/put-block it's weird that you didn't get an error, can you take a look what's going wrong there?

Reply:
Whoops Idk why I used metadata for testing, I have fixed that.
As for the reason the metadata is found in headers its because of this https://github.com/Azure/azure-sdk-for-python/blob/c9c98c2f5c113055c3777e51c55c6b0e573e9947/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L374 
It appears that metadata is sent as part of the headers but we append a prefix `x-ms-meta-` for the service to know that this is metadata. But it seems the PUT requests does accept `x-ms-meta-` in the headers, this is proven by this line: https://github.com/Azure/azure-sdk-for-python/blob/c9c98c2f5c113055c3777e51c55c6b0e573e9947/sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py#L150 `commit_block_list` is a PUT request that always took the headers including the metadata.
So that means with the changes I made in this PR the metadata would be sent in the headers with every PUT request.

Reply:
While the user is complaining that the stage block operation doesn't have the customized headers:
https://github.com/Azure/azure-sdk-for-python/blob/c9c98c2f5c113055c3777e51c55c6b0e573e9947/sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py#L132
and 
https://github.com/Azure/azure-sdk-for-python/blob/c9c98c2f5c113055c3777e51c55c6b0e573e9947/sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py#L120
Not the two places you mentioned.

Reply:
Offline discussion conclusion: service side ignores headers it doesn't accept, `x-ms-meta` in this case